### PR TITLE
[sync] 2022/06/06

### PR DESCRIPTION
**Pull request by AlexV525**
#123

### CODE REVIEW

File: src/_includes/cookie-notice.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<section id="cookie-notice">
```
Comment (AlexV525) [inline src/_includes/cookie-notice.html:1]: FYI @chenglu for extra notice

File: src/_guides/language/language-tour.md [hunk 2]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+* All [generative constructors](#constant-constructors) must be constant.
+* [Factory constructors](#factory-constructors) can only return
+  one of the fixed, known enum instances.
+* No other class can be extended as [`Enum`] is automatically extended.
+* There cannot be overrides for `index`, `hashCode`, the equality operator `==`.
+* A member named `values` cannot be declared in an enum,
+  as it would conflict with the automatically generated static `values` getter.
+* All instances of the enum must be declared
+  in the beginning of the declaration,
+  and there must be at least one instance declared.
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4926]:
```suggestion
* All instances of the enum must be declared
  in the beginning of the declaration,
  and there must be at least one instance declared.

  在进行枚举定义时，所有的实例都需要首先进行声明，
  且至少要声明一个枚举实例。

```

File: src/_guides/language/language-tour.md [hunk 3]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+
+Here is an example that declares an enhanced enum
+with multiple instances, instance variables,
+a getter, and an implemented interface:
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4930]:
```suggestion
Here is an example that declares an enhanced enum
with multiple instances, instance variables,
a getter, and an implemented interface:

下方是一个增强枚举的例子，
它包含多个枚举实例、成员变量、getter 并且实现了接口：
```

File: src/_guides/language/language-tour.md [hunk 4]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+
+  int get carbonFootprint => (carbonPerKilometer / passengers).round();
+
+  @override
+  int compareTo(Vehicle other) => carbonFootprint - other.carbonFootprint;
+}
+```
+
+To learn more about declaring enhanced enums,
+see the section on [Classes](#classes).
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4957]:
```suggestion
To learn more about declaring enhanced enums,
see the section on [Classes](#classes).

想要了解更多关于定义增强枚举的内容，
可以阅读 [类](#classes) 小节。
```

File: src/_guides/language/language-tour.md [hunk 5]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+
+{{site.alert.version-note}}
+  Enhanced enums require a [language version][] of at least 2.17.
+{{site.alert.end}}
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4961]:
```suggestion
{{site.alert.version-note}}

  Enhanced enums require a [language version][] of at least 2.17.

  增强枚举仅在 [Dart SDK 版本][language version] 2.17 以上可用。

{{site.alert.end}}
```

File: src/_guides/language/language-tour.md [hunk 6]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+
+#### Using enums
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4963]:
```suggestion
#### Using enums

#### 使用枚举
```

File: src/_guides/language/language-tour.md [hunk 7]
```diff
@@ -4778,21 +4863,115 @@ a fixed number of constant values.
...
+
+Access the enumerated values like
+any other [static variable](#static-variables):
```
Comment (AlexV525) [inline src/_guides/language/language-tour.md:4966]:
```suggestion
Access the enumerated values like
any other [static variable](#static-variables):

你可以像访问 [静态变量](#static-variables) 一样访问枚举值：
```

File: src/get-dart/index.md [hunk 8]
```diff
@@ -71,7 +71,7 @@ Dart SDK 支持 Windows、Linux 和 macOS。
 
 * **Supported versions:** Windows 10.
```
Comment (chenglu) [inline src/get-dart/index.md:72]:
```suggestion
* **Supported versions:** Windows 10 and 11.
```

File: src/_layouts/homepage.html [hunk 9]
```diff
@@ -2,6 +2,7 @@
 <html>
   {% include head.html %}
   <body class="homepage">
+    {% include cookie-notice.html %}
```
Comment (chenglu) [inline src/_layouts/homepage.html:5]:
```suggestion
    <!-- {% include cookie-notice.html %} -->
```

File: src/_layouts/default.html [hunk 10]
```diff
@@ -2,6 +2,7 @@
 <html>
   {% include head.html %}
   <body class="{{page.layout}}{% if page.toc == false %} hide_toc{% endif %}{% if page.obsolete == true %} obsolete{% endif %}{% if site.show_banner == true %} show_banner{% endif %}">
+    {% include cookie-notice.html %}
```
Comment (chenglu) [inline src/_layouts/default.html:5]:
```suggestion
    <!-- {% include cookie-notice.html %} -->
```

**Review by chenglu — APPROVED** [review]
LGTM

**Pull request merged**